src/jitlayers.h

```diff
@@ -55,14 +55,6 @@
 #endif
 // The sanitizers don't play well with our memory manager
 
-#if defined(JL_FORCE_JITLINK) || defined(_CPU_AARCH64_) || defined(HAS_SANITIZER)
```

Comment:
Maybe merge this separately, so we can get feedback if this causes issues for anyone separately from the PR (and because github has a hard time dealing with a diff this moderately large)